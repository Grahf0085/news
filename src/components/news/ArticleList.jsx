import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleElements = articles.map((article) => (
    <li key={article.title}>
      <Article 
        title={article.title}
        author={article.author}
        description={article.description}
      />
    </li>
  ));

  return <ul data-testid="list">{articleElements}</ul>;

};

ArticleList.propTypes = {
  articles: PropTypes.objectOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired
};

export default ArticleList;
