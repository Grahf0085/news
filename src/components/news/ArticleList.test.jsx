import React from 'react';
import { render } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList component', () => {
  it('renders title, author, description', () => {
    const articles = [{
      title: 'olympics are so cool',
      author: 'some person',
      description: 'people scream a lot'
    }];

    const { asFragment } = render(<ArticleList articles={articles}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
