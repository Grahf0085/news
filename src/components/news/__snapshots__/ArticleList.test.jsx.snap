// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders title, author, description 1`] = `
<DocumentFragment>
  <ul
    data-testid="list"
  >
    <li>
      <section
        data-testid="display"
      >
        <p>
          Title: olympics are so cool
        </p>
        <p>
          Author: some person
        </p>
        <p>
          Description: people scream a lot
        </p>
      </section>
    </li>
  </ul>
</DocumentFragment>
`;
